pkg/cmd/repo/rename/http.go

```diff
@@ -0,0 +1,40 @@
+package rename
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/internal/ghinstance"
+	"github.com/cli/cli/v2/internal/ghrepo"
+)
+
+func runRename(client *http.Client, repo ghrepo.Interface, newRepoName string) error {
+	input := map[string]string{"name": newRepoName}
+	body, err := json.Marshal(input)
+	if err != nil {
+		return err
+	}
+
+	path := fmt.Sprintf("%srepos/%s",
+		ghinstance.RESTPrefix(repo.RepoHost()),
+		ghrepo.FullName(repo))
+
+	request, err := http.NewRequest("PATCH", path, bytes.NewBuffer(body))
+	if err != nil {
+		return err
+	}
+
+	response, err := client.Do(request)
+	if err != nil {
+		return err
+	}
+	defer response.Body.Close()
+
+	if response.StatusCode > 299 {
+		return api.HandleHTTPError(response)
+	}
+	return nil
```

Comment:
Instead of returning just an error object, I would suggest that this function should parse the JSON response from the server (which will be information about the Repository) and construct a `ghrepo.Interface` value from the final name of the repository. This is to give the caller the ability to print the actual name of the repository after the rename: it might be different than the newRepoName print argument due to repository name normalization that GitHub performs.

For example: I could call `gh repo rename -R mislav/playground 'my new! playground.js.git'`, but that isn't going to rename the repo to `mislav/my new! playground.js.git`, since that is an invalid GitHub repository name. Instead, GitHub will name it `mislav/my-new-playground.js`. So, the only way to know the final repo name with 100% accuracy is not to assume what the final repo name will be based on user's input, but to read the response from the server after the rename and print that.